extensions/azurecore/src/account-provider/utils/msalCachePlugin.ts

```diff
@@ -18,35 +19,53 @@ export class MsalCachePluginProvider {
 		Logger.verbose(`MsalCachePluginProvider: Using cache path ${_msalFilePath} and serviceName ${_serviceName}`);
 	}
 
+	private getLockfilePath(): string {
+		return this._msalFilePath + '.lock';
+	}
+
 	public getCachePlugin(): ICachePlugin {
+		const lockFilePath = this.getLockfilePath();
 		const beforeCacheAccess = async (cacheContext: TokenCacheContext): Promise<void> => {
-			let exists = true;
+			let isAccessible = true;
+			await this.waitAndLock(lockFilePath);
 			try {
-				await fsPromises.access(this._msalFilePath, constants.R_OK | constants.W_OK);
-			} catch {
-				exists = false;
-			}
-			if (exists) {
 				try {
+					await fsPromises.access(this._msalFilePath, constants.R_OK | constants.W_OK);
```

Comment:
Generally checking file access and then immediately trying to read it is considered an anti-pattern, since things can change between when you do the initial stat and when you try to read the file. It's usually better/safer to just instead try to read the file and then handle any errors that occur during the read at that point (in this case just logging and error and moving on

Reply:
Makes sense, I too was confused whether is call is needed, will remove it.